File: Assets/Recipes/TinyArmsAttack/Scripts/MovementSystem.ts

namespace game {

    /** New System */
    export class MovementSystem extends ut.ComponentSystem {
        
        OnUpdate(): void
        {
            let dt = this.scheduler.deltaTime();

            this.world.forEach([ut.Core2D.TransformLocalPosition, game.Movement], (transformLocal, movement) => {

                //TODO: Change this to a tag
                if (!movement.isMoving)
                    return;

                let position = transformLocal.position;

                position.x += movement.speed * dt;

                console.log(movement.speed);

                transformLocal.position = position;
            });
        }
    }
}

File: Assets/Recipes/TinyTinyArms/Scripts/MovementSystem.cs
using Unity.Entities;
using Unity.Tiny.Core;
using Unity.Tiny.Core2D;

public class MovementSystem : ComponentSystem
{
    protected override void OnUpdate()
    {
        var dt = (float)World.TinyEnvironment().frameDeltaTime;

        Entities.ForEach((ref Translation translation, ref Movement movement) =>
        {
            translation.Value.x += movement.speed * dt;
        });
    }
}
